tests/run/legacy_implicit_noexcept.pyx

```diff
@@ -0,0 +1,138 @@
+# cython: legacy_implicit_noexcept=True
+# mode: run
+# tag: warnings
+import cython
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+cdef int func_implicit(int a, int b):
+    raise RuntimeError
+
+cdef int func_noexcept(int a, int b) noexcept:
+    raise RuntimeError
+
+cdef int func_star(int a, int b) except *:
+    raise RuntimeError
+
+cdef int func_value(int a, int b) except -1:
+    raise RuntimeError
+
+cdef func_return_obj_implicit(int a, int b):
+    raise RuntimeError
+
+cdef int(*ptr_func_implicit)(int, int)
+ptr_func_implicit = func_implicit
+
+cdef int(*ptr_func_noexcept)(int, int) noexcept
+ptr_func_noexcept = func_noexcept
+
+@cython.cfunc
+def pure_func_implicit():
+    raise RuntimeError
```

Comment:
The more interesting case is probably `@cython.cfunc @cython.returns(int)` where the behaviour might be different. I'm not sure off the top of my head what the 0.29.x and current master behaviour is though. But I have a feeling we might not be dealing with this right.

If we aren't dealing with it right then this PR probably isn't the place to do it though.

Reply:
Eh I missed to return cython.int. You are explicitely interesed in `@cython.returns(int)` or just for case returning int? (`-> cython.int`)

Reply:
Just the case returning a C integer. So using the `->` syntax is probably better.

Reply:
Ok great. So it doesn't really change anything. Probably worth having the tests but no need to go into too much more detail there